app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+class PostsController < ApplicationController
+  include PostsHelper
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    if signed_in?
+      @post = current_user.posts.new
+    else
+      flash[:error] = 'Please, sign in to create post'
+      redirect_to new_session_path
+    end
+  end
+
+  def create
```

Comment:
We are not putting any security/authorization for creating posts? We should provide a :before_action callback that would check first if the user is authorized before creating a post. This same callback can be used for the `new` action.

Reply:
According to the project instructions, all users can create a post. As long as the author's name doesn't show when the user is logged out.

Reply:
Yes it does. However since this kind of feature will be common in all the apps you will build it is a good practice to secure creation of posts. It's a simple fix -> just create a private method in your controller that returns a boolean depending on whether a user is authenticated then you will just use it in `before_action only[new, create]` where if it returns false then you will redirect the user to the login page.

Reply:
Alright @rvvergara, we've adjusted accordingly. Thanks for the explicit explanation.